cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AuthorizationHandler.java

```diff
@@ -62,14 +78,17 @@
     this.enabled = conf.getBoolean(Constants.Security.Authorization.ENABLED);
   }
 
-  private void createLogEntry(HttpRequest httpRequest, AuthorizationRequest request,
+  private void createLogEntry(HttpRequest httpRequest, @Nullable AuthorizationRequest request,
                               HttpResponseStatus responseStatus) throws UnknownHostException {
-    String reqBody = String.format("[%s %s %s]", request.getPrincipal(), request.getEntity(), request.getActions());
+
     AuditLogEntry logEntry = new AuditLogEntry();
     logEntry.setUserName(Objects.firstNonNull(SecurityRequestContext.getUserId(), "-"));
     logEntry.setClientIP(InetAddress.getByName(Objects.firstNonNull(SecurityRequestContext.getUserIP(), "0.0.0.0")));
     logEntry.setRequestLine(httpRequest.getMethod(), httpRequest.getUri(), httpRequest.getProtocolVersion());
-    logEntry.setRequestBody(reqBody);
+    if (request != null) {
+      logEntry.setRequestBody(String.format("[%s %s %s]", request.getPrincipal(), request.getEntity(),
```

Comment:
I don't think it will happen. Their respective toString() will be called. Here is a sample which I got while running unit tests. Looks good to me. 

0.0.0.0 - - [17/Mar/2016:23:10:49 -0700] "POST /v3/security/authorization/grant HTTP/1.1" [Principal{name='admin', type=ROLE} namespace:ns1 [READ]] 200 -